### PR TITLE
feat: #4342 adds new menu and toggles to geo diary

### PR DESCRIPTION
**Pull request by willmcvay**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #4342

### CODE REVIEW

File: packages/geo-diary/src/components/pages/appointment/__styles__/index.ts [hunk 1]
```diff
@@ -5,28 +5,28 @@ export const MapContainer = styled.div`
   overflow: hidden;
   position: absolute;
   width: 100%;
-  height: calc(100% - 15.6rem);
-  top: 15.6rem;
+  height: calc(100% - 15rem);
+  top: 15rem;
```
Comment (bashleigh) [inline packages/geo-diary/src/components/pages/appointment/__styles__/index.ts:9]: constants? :p

File: packages/geo-diary/src/components/pages/appointment/__styles__/index.ts [hunk 2]
```diff
@@ -5,28 +5,28 @@ export const MapContainer = styled.div`
...
 
   @media (min-width: 769px) {
     top: 0;
     height: 100%;
-    left: calc(390px + 4.6rem);
-    width: calc(100vw - 29rem);
+    left: 467px;
+    width: calc(100vw - 467px);
```
Comment (bashleigh) [inline packages/geo-diary/src/components/pages/appointment/__styles__/index.ts:15]: more variables?

File: packages/geo-diary/src/components/pages/appointment/__styles__/index.ts [hunk 3]
```diff
@@ -35,18 +35,18 @@ export const AppoinmentContainer = styled.div`
   padding: 1.5rem 1.5rem;
   position: absolute;
   background: #f5f7f9;
-  height: calc(100% - 15.6rem);
+  height: calc(100% - 15rem);
```
Comment (bashleigh) [inline packages/geo-diary/src/components/pages/appointment/__styles__/index.ts:38]: var?

File: packages/geo-diary/src/components/pages/appointment/__styles__/index.ts [hunk 4]
```diff
@@ -35,18 +35,18 @@ export const AppoinmentContainer = styled.div`
...
 
   @media (min-width: 769px) {
-    width: 25rem;
-    left: 4.1rem;
-    top: 11.6rem;
-    height: calc(100% - 11.6rem);
+    width: 24.1rem;
+    left: 5rem;
+    top: 11.5rem;
+    height: calc(100% - 11.5rem);
```
Comment (bashleigh) [inline packages/geo-diary/src/components/pages/appointment/__styles__/index.ts:49]: more var?

File: packages/geo-diary/src/components/ui/appointment-time/appointment-time.tsx [hunk 5]
```diff
@@ -21,29 +21,34 @@ export const AppointmentTime = () => {
...
+              id: 'tomorrow',
+              value: 'TOMORROW',
+              text: 'Tomorrow',
+              isChecked: time === 'TOMORROW',
+            },
+            {
+              id: 'week',
+              value: 'WEEK',
+              text: 'Week',
+              isChecked: time === 'WEEK',
```
Comment (bashleigh) [inline packages/geo-diary/src/components/ui/appointment-time/appointment-time.tsx:46]: use enum for time values? Easier for type hinting and usage

```ts
enum AppTimeRange {
  TODAY = 'today',
  TOMORROM = 'tomorrow',
  WEEK = 'week',
}

const options: {
  id: keyof AppTimeRange,
  value: AppTimeRange,
  text: string,
  isChecked: boolean
}[] = Object.keys(AppTimeRange).map(timeRange => ({  id: timeRange, value: AppTimeRange[timeRange], text: timeRange.capitalize(), isChecked: time === AppTimeRange[timeRange] })
```

types vs enum: types such as `type AppTimeRange = 'TODAY' | 'TOMORROW'` cannot be used as a value. Where as an enum can be a type and a value.

File: packages/geo-diary/src/components/ui/nav/nav.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React from 'react'
+import { useReapitConnect } from '@reapit/connect-session'
+import { reapitConnectBrowserSession } from '@/core/connect-session'
+import { NavResponsive, NavResponsiveOption } from '@reapit/elements/v3'
+
+export const callbackAppClick = () =>
+  (window.location.href =
+    window.location.href.includes('dev') || window.location.href.includes('localhost')
+      ? 'https://marketplace.dev.paas.reapit.cloud/installed'
+      : 'https://marketplace.reapit.cloud/installed')
```
Comment (bashleigh) [inline packages/geo-diary/src/components/ui/nav/nav.tsx:10]: constants? :p

File: packages/geo-diary/src/components/ui/tab-mode/tab-mode.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+              id: 'diary',
+              value: 'LIST',
+              text: 'Diary',
+              isChecked: tab === 'LIST',
+            },
+            {
+              id: 'map',
+              value: 'MAP',
+              text: 'Map',
+              isChecked: tab === 'MAP',
```
Comment (bashleigh) [inline packages/geo-diary/src/components/ui/tab-mode/tab-mode.tsx:41]: see enum suggestion

File: packages/geo-diary/src/components/ui/travel-mode/travel-mode.tsx [hunk 8]
```diff
@@ -20,22 +22,28 @@ export const TravelMode: FC = () => {
...
+              value: 'DRIVING',
+              text: 'Driving',
+              isChecked: travelMode === 'DRIVING',
+            },
+            {
+              id: 'walking',
+              value: 'WALKING',
+              text: 'Walking',
+              isChecked: travelMode === 'WALKING',
+            },
```
Comment (bashleigh) [inline packages/geo-diary/src/components/ui/travel-mode/travel-mode.tsx:42]: see enum suggestion

**Review by bashleigh — CHANGES_REQUESTED** [review]
Looks good! Just some comments and suggestions about enum usage and constants

File: packages/geo-diary/src/components/ui/appointment-time/appointment-time.tsx (hunk 5 above)
Comment (willmcvay) [reply packages/geo-diary/src/components/ui/appointment-time/appointment-time.tsx:46]: As discussed, an enum may not be appropriate here and would be a big refactor

File: packages/geo-diary/src/components/pages/appointment/__styles__/index.ts [hunk 9]
```diff
@@ -1,32 +1,39 @@
 import { css } from 'linaria'
 import { styled } from 'linaria/react'
 
+const navControlsHeightMobile = '15rem'
+const navHeightMobile = '3.5rem'
+const navAppointmentListWidthDesktop = '467px'
+const controlsHeightDesktop = '9.1rem'
+const navWidthDesktop = '5rem'
+const appointmentListWidthDesktop = '24.1rem'
```
Comment (bashleigh) [inline packages/geo-diary/src/components/pages/appointment/__styles__/index.ts:9]: <3

**Review by bashleigh — APPROVED** [review]

**Pull request merged**